packages/webpack-cli/src/webpack-cli.ts

```diff
@@ -2363,16 +2364,30 @@ class WebpackCLI implements IWebpackCLI {
     options: Partial<WebpackDevServerOptions>,
     callback?: Callback<[Error | undefined, WebpackCLIStats | undefined]>,
   ): Promise<WebpackCompiler> {
-    if (typeof options.defineProcessEnvNodeEnv === "string") {
-      // TODO: This should only set NODE_ENV for the runtime not for the config too. Change this during next breaking change.
-      process.env.NODE_ENV = options.defineProcessEnvNodeEnv;
-    } else if (typeof options.nodeEnv === "string") {
+    const isNodeEnvOptionDefined = typeof options.nodeEnv === "string";
+    const isDefineProcessEnvNodeEnvOptionDefined =
+      typeof options.defineProcessEnvNodeEnv === "string";
```

Comment:
@snitin315 I don't think we need this option anymore anyway. We have - https://github.com/webpack/webpack/blob/main/lib/WebpackOptionsApply.js#L599.

The main problem is that we have two options:
- `--node-env` (set `process.env.NODE_ENV` for `webpack.config.js`)
- `--optimization-node-env` (set `process.env.NODE_ENV` for runtime)

And that's why they confuse some developers. The names are almost the same, but they do completely different things.

I think the best solution would be to do the following:
- remove `--define-process-env-node-env` at all
- we leave `--node-env`  because it is widely used by many  
- we come up with a new name for the `--node-env` variable and updates the documentation so that new developers can use it

Shortly `--node-env` and `---new-name-node-env` work and do the same thing, but the documentation will recommend to use a new name.

To be honest I have no ideas for a new name, so I'd be happy to hear any suggestions :smile:

Reply:
@alexander-akait how about `--config-node-env`?

Reply:
`--config-node-env` sounds good

Reply:
Updated 👍🏻